airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/JobCreationAndStatusUpdateActivityImpl.java

```diff
@@ -261,6 +265,42 @@ public void reportJobStart(final ReportJobStartInput input) {
     }
   }
 
+  @Override
+  public void failNonTerminalJobs(final FailNonTerminalJobsInput input) {
+    log.info("failNonTerminalJobs called with input: {}", input);
+    try {
+      final Set<io.airbyte.scheduler.models.JobStatus> nonTerminalJobStatuses = Sets.difference(
+          Set.of(io.airbyte.scheduler.models.JobStatus.values()),
+          io.airbyte.scheduler.models.JobStatus.TERMINAL_STATUSES);
+      final List<Job> jobs = jobPersistence.listJobsForConnectionWithStatuses(input.getConnectionId(), Job.REPLICATION_TYPES, nonTerminalJobStatuses);
+      log.info("failNonTerminalJobs jobs result: {}", jobs);
+      for (final Job job : jobs) {
+        final long jobId = job.getId();
+        log.info("Failing non-terminal job {}", jobId);
+        jobPersistence.failJob(jobId);
+
+        // fail all non-terminal attempts
+        for (final Attempt attempt : job.getAttempts()) {
+          if (Attempt.isAttemptInTerminalState(attempt)) {
+            continue;
+          }
+
+          // the Attempt object 'id' is actually the value of the attempt_number column in the db
+          final int attemptNumber = (int) attempt.getId();
```

Comment:
That sad that this confusion is still here. It is very bad:
in the DB it is an int that we cast into a long in java and then we cast it back to an int plus it is not the id but the number 🤦 . Nothing that can be done in this PR unfortunately. Thanks for adding a comment to explain.

Reply:
Yeah it is definitely not ideal. Maybe we should have a separate migration to rename this column from `id` to `attempt_number`?

Reply:
The column is named `attempt_number` but it is put in the id field. You can see it in `getAttemptFromRecord` in `DefaultJobPersistence`. We can create a ticket for it but it can be quite dangerous if we miss one update.

Reply:
Ah, that actually might technically be an easier change then. But yeah we will need to make sure we update every place that uses the `Attempt.id` field, which could be quite a few places.